parsing/g_prim.mlg

```diff
@@ -120,26 +119,26 @@ GRAMMAR EXTEND Gram
   lstring:
     [ [ s = string -> { CAst.make ~loc s } ] ]
   ;
+  caml_integer:
+    [ [ i = integer -> { my_int_of_string ~loc i } ] ]
+  ;
+  caml_natural:
+    [ [ i = natural -> { my_int_of_string ~loc i } ] ]
```

Comment:
```suggestion
    [ [ i = natural -> { nat63_of_string ~loc i } ] ]
```

The check was incorrect--was checking for an int, not a nat.  And suggested renaming.

For the various types of numbers, probably clearer to go directly to `NUMBER` instead of cascading nonterminals.  `common.edit_mlg` has to rewrite these productions anyway.